vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,603 @@
+---
+title: "Demonstration of netCDF functionality in rSW2analysis"
+author: "Caitlin M. Andrews"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+    toc_depth: 2
+vignette: >
+  %\VignetteIndexEntry{Demonstration of netCDF functionality in rSW2analysis}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Overview
+
+This vignette is an overview of how to use the functions in the rSW2analysis
+package to convert arrays of rSOILWAT2 and SOILWAT2 output data into netCDFs
+that adhere to
+[CF conventions 1.8](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html)
+and [CMIP6 conventions](https://docs.google.com/document/d/1h0r8RZr_f3-8egBMMh7aqLwy3snpD6_MrDz1q8n5XUk/edit#).
+
+Additionally, this document describes the Dryland Ecology Lab's standards and
+conventions in regards to naming, metadata, and netCDF structure.
+
+# Input Data Format and Structure
+
+## Variable Data
+
+The netCDF functions in this package expect an array, in which each row
+represents data for a site or a gridcell. Columns represent one of three things:
+
+* multiple different variables for one site
+* one variable with multiple measurements (time series)
+* one variable with a vertical component, e.g., multiple soil depth measurements
+
+```{r, create1d, results = "hide"}
+# load rSW2analysis package
+library(rSW2analysis)
+
+# dummy data for 10 sites, 10 data points (could be 10 vars, times, or depths)
+someData <- rnorm(100, 7, 30)
+data2d <- array(someData, c(10, 10))
+str(data2d)
+is.array(data2d)
+```
+
+Additionally, it is possible to have data at multiple times *and* multiple depths.
+In this case, time should be organized by column, and multiple depths should be organized
+sequentially in third dimension, forming a 3d array.
+
+```{r, create3d, results = "hide"}
+# dummy data for 10 sites, 10 times, 3 depths.
+data3d <- array(0, c(10, 10, 3))
+data3d[, , 1] <- someData
+data3d[, , 2] <- someData
+data3d[, , 3] <- someData
+```
+
+## Location Data
+
+Locations all need to be on the same coordinates reference system (CRS). Data can
+be either be organized on a regularly spaced grid (e.g. wall to wall simulations for
+a given area) or not (e.g. project data representing plot points in a park).
+
+If the data is on a grid, either a gridded raster file or a table of coordinates
+can be provided. If the former, the coordinates will be extracted from the grid.
+If the later, the coordinates need to be ordered from least to greatest. For example,
```

Comment:
I'm not sure what "coordinates ... ordered from least to greatest" means. Since these are numbers maybe from smallest to largest, but probably more usual and useful would be to remind that the orientation is east (for longitude) and north (for latitude), e.g., https://epsg.org/crs_4326/WGS-84.html and https://en.wikipedia.org/wiki/Longitude, and should be ordered accordingly?

Reply:
Tried to clarify